### PR TITLE
HARVESTER: fix auto choose cd-rom issue

### PR DESCRIPTION
**Pull request by WuJun2016**
# links
- https://github.com/harvester/harvester/issues/2057

### CODE REVIEW

File: edit/harvesterhci.io.virtualmachineimage.vue [hunk 1]
```diff
@@ -80,6 +87,30 @@ export default {
 
       this.value.spec.url = url;
       if (VM_IMAGE_FILE_FORMAT.includes(fileSuffiic)) {
+        if (fileSuffiic) {
```
Comment (n313893254) [inline edit/harvesterhci.io.virtualmachineimage.vue:90]: fileSuffiic won't be null becauseof `VM_IMAGE_FILE_FORMAT.includes(fileSuffiic)`

Comment (n313893254) [inline edit/harvesterhci.io.virtualmachineimage.vue:90]: fileSuffiic won't be null becauseof `VM_IMAGE_FILE_FORMAT.includes(fileSuffiic)`

Comment (n313893254) [reply edit/harvesterhci.io.virtualmachineimage.vue:90]: Also spelling should be `fileSuffix`

File: edit/harvesterhci.io.virtualmachineimage.vue [hunk 2]
```diff
@@ -80,6 +87,30 @@ export default {
...
+          const os = this.getOSType(url);
+
+          if (os) {
+            Object.keys(this.value.labels).forEach((key, idx) => {
+              if (key === HCI_ANNOTATIONS.OS_TYPE) {
+                this.$refs.labels.remove(idx);
+              }
+            });
+            this.$refs.labels.add(HCI_ANNOTATIONS.OS_TYPE, os.value);
+          }
```
Comment (n313893254) [inline edit/harvesterhci.io.virtualmachineimage.vue:111]: Is the os has reasons to put it under the `if (VM_IMAGE_FILE_FORMAT.includes(fileSuffiic))` condition?

Comment (n313893254) [inline edit/harvesterhci.io.virtualmachineimage.vue:111]: Is the os has reasons to put it under the `if (VM_IMAGE_FILE_FORMAT.includes(fileSuffiic))` condition?

**Review by n313893254 — CHANGES_REQUESTED** [review]

File: edit/harvesterhci.io.virtualmachineimage.vue [hunk 3]
```diff
@@ -144,6 +175,46 @@ export default {
...
+    },
+
+    getOSType(url) {
+      if (!url) {
+        return;
+      }
+
+      return OS.find( (os) => {
+        if (os.match) {
+          return os.match.find(mactchValue => url.toLowerCase().includes(mactchValue)) ? os.value : false;
```
Comment (n313893254) [inline edit/harvesterhci.io.virtualmachineimage.vue:212]: `mactch` is spelling error

**Review by n313893254 — CHANGES_REQUESTED** [review]
Recommend using Code Spell Checker to decrease spelling errors, but the plugin dictionary is so less, it will throw many unknown word errors.....

image

File: edit/harvesterhci.io.virtualmachineimage.vue (hunk 1 above)
Comment (WuJun2016) [reply edit/harvesterhci.io.virtualmachineimage.vue]: Done!

File: edit/harvesterhci.io.virtualmachineimage.vue (hunk 2 above)
Comment (WuJun2016) [reply edit/harvesterhci.io.virtualmachineimage.vue]: Done.

Comment (WuJun2016) [reply edit/harvesterhci.io.virtualmachineimage.vue]: Done

File: edit/harvesterhci.io.virtualmachineimage.vue (hunk 3 above)
Comment (WuJun2016) [reply edit/harvesterhci.io.virtualmachineimage.vue]: Done

**Comment by WuJun2016** [comment]
done,  PTAL, thanks

**Review by n313893254 — APPROVED** [review]

**Pull request merged**